distribution/push_v2.go

```diff
@@ -180,8 +182,29 @@ func (p *v2Pusher) pushV2Tag(ctx context.Context, ref reference.NamedTagged, id
 
 	putOptions := []distribution.ManifestServiceOption{distribution.WithTag(ref.Tag())}
 	if _, err = manSvc.Put(ctx, manifest, putOptions...); err != nil {
-		logrus.Warnf("failed to upload schema2 manifest: %v", err)
-		return err
+		if runtime.GOOS == "windows" || p.config.TrustKey == nil || p.config.RequireSchema2 {
+			logrus.Warnf("failed to upload schema2 manifest: %v", err)
+			return err
+		}
+
+		logrus.Warnf("failed to upload schema2 manifest: %v - falling back to schema1", err)
+
+		logrus.Warnf("[DEPRECATION NOTICE] schema1 support will be removed in an upcoming release. Please contact admins of the %s registry NOW to avoid future disruption. Waiting 13 annoying seconds...", reference.Domain(ref))
+		time.Sleep(13 * time.Second)
```

Comment:
to be debated

Reply:
We could randomize it?

If this needs more discussion, might be good to move it to a separate PR to unblock this one